.github/workflows/frontend_charm_integration_tests.yml

```diff
@@ -1,24 +1,52 @@
 name: Run charm tests
 on:
   push:
-    branches: ["main"]
+    branches: ["main", "integration-test-fix*"]
     tags: ["v*.*.*"]
 
 jobs:
   integration-test:
+    permissions:
+      contents: read
+      packages: read
+
     name: Integration tests
-    runs-on: ubuntu-22.04
+    runs-on: [self-hosted, jammy, edge]
     steps:
       - name: Checkout
         uses: actions/checkout@v3
+      - name: Set up docker buildx
+        uses: docker/setup-buildx-action@v3
+        with:
+          config-inline: |
+            [registry."docker.io"]
+              mirrors = ["https://github-runner-dockerhub-cache.canonical.com:5000"]
       - name: Setup LXD
         uses: canonical/setup-lxd@main
         with:
-          channel: 5.13/stable
+          channel: 5.19/stable
       - name: Setup operator environment
-        uses: charmed-kubernetes/actions-operator@main
+        uses: mz2/actions-operator@4fcb55b99e2a0f69a5910870d3639d203aeb87aa
```

Comment:
Because otherwise I can't force strictly confined versions of juju and microk8s to be used.

Reply:
I have posted a PR with my changes, indeed should be able to merge back.

For clarity this branch when in a mergeable state will be the first point at which these charm integration tests actually behave.

Reply:
Plot twist: turns out my tweaks here were unnecessary because snap ignores `--classic` for a strictly confined snap: https://github.com/charmed-kubernetes/actions-operator/pull/57 --> I have changed this back to the way it was before.